packages/project-access/src/project/search.ts

```diff
@@ -407,3 +407,33 @@ export async function findFioriArtifacts(options: {
     }
     return results;
 }
+
+/**
+ * Find all CAP project roots by locating pom.xml or package.json in a given workspace.
+ *
+ * @param options - find options
+ * @param options.wsFolders - list of roots, either as vscode WorkspaceFolder[] or array of paths
+ * @returns - root file paths that may contain a CAP project
+ */
+export async function findCapProjects(options: {
+    readonly wsFolders: WorkspaceFolder[] | string[];
+}): Promise<string[]> {
+    const result = new Set<string>();
+    const excludeFolders = ['node_modules', 'dist', 'src', 'webapp', 'MDKModule', 'gen'];
+    const fileNames = [FileName.Pom, FileName.Package];
+    const wsRoots = wsFoldersToRootPaths(options.wsFolders);
+    for (const root of wsRoots) {
+        const filesToCheck = await findBy({
+            fileNames,
+            root,
+            excludeFolders
+        });
+        const foldersToCheck = Array.from(new Set(filesToCheck.map((file) => dirname(file)))); //only directories without duplicates
+        for (const folderToCheck of foldersToCheck) {
+            if ((await getCapProjectType(folderToCheck)) !== undefined) {
```

Comment:
Can we add additional validation, perhaps optionally, to the function `getCapProjectType` as the current checks are too rudimentary to be useful in some cases. We need to also validate that the services path exists and is accessible (correct file permissions), otherwise we will return non-cap projects that will not be compilable.

Reply:
@IainSAP, good call, I've ran against https://github.com/SAP-samples/cloud-cap-samples and it found the root of the repo as CAP project. Added check for `srv` folder: https://github.com/SAP/open-ux-tools/pull/1212/commits/d7045e3ee1ee582c2a89afc299888a66444e911d. Will add some more test cases and unit tests.

Reply:
Tests modified/added

Reply:
Could we add logs (and allow an external logger to be passed) so we can trace skipped projects and why they are skipped. This is needed so we can easily debug issues with CAP discovery.